tests/integration/tests/test_util/k8s.py

```diff
@@ -0,0 +1,260 @@
+#
+# Copyright 2025 Canonical, Ltd.
+#
+import json
+import logging
+from fnmatch import fnmatch
+from typing import Any, Dict, List, Optional
+
+from test_util import harness
+
+LOG = logging.getLogger(__name__)
+
+
+def get_resources_in_namespaces(
```

Comment:
The naming feels a bit weird/unfit here, if I understand correctly we are mainly dealing with `apps` type resources and specifically check for containers etc. I assume we don't want this to be used for services, ingress etc. right? Maybe we should update this to a name that reflects it better?